src/Network/Minio/Data.hs

```diff
@@ -1001,6 +1002,35 @@ type Stats = Progress
 -- Select API Related Types End
 --------------------------------------------------------------------------
 
+----------------------------------------------
+-- Bucket Versioning Related Types
+----------------------------------------------
+
+data BucketVersioning
+  = BVDisabled
+  | BVSuspended
+  | BVEnabled
+  deriving stock (Eq, Show)
+
+newtype MFAToken = MFAToken {unMfaToken :: Text}
+  deriving stock (Eq, Show)
+  deriving (IsString) via Text
```

Comment:
mfatoken is just a text, so we can use newtype deriving here:

```hs
  deriving stock (Show)
  deriving newtype (Eq, IsString)
```

Reply:
Actually looks like MFAToken is just a sort of Boolean - https://docs.aws.amazon.com/AmazonS3/latest/API/API_GetBucketVersioning.html

we can just do:

```hs
data MFADelete = MFADeleteEnabled | MFADeleteDisabled
```